opentelemetry-android-instrumentation/src/main/java/io/opentelemetry/rum/internal/instrumentation/activity/ActivityTracer.java

```diff
@@ -195,13 +198,22 @@ private String getActivityName() {
             return activity.getClass().getSimpleName();
         }
 
-        public ActivityTracer build() {
-            return new ActivityTracer(this);
-        }
-
         public Builder setScreenName(String screenName) {
             this.screenName = screenName;
             return this;
         }
+
+        public ActivityTracer build() {
+            if (activeSpan == INVALID_ACTIVE_SPAN) {
+                throw new IllegalStateException("activeSpan must be configured.");
+            }
```

Comment:
Why not `if (activeSpan == null)`?
I'm guessing that'd require adding some `@Nullable` and `requireNonNull()`, but isn't that less bad than creating fake invalid instances/

Reply:
Let's see what it looks like....

Reply:
Bleh... welp, too bad, at least we tried.